src/agent.cpp

```diff
@@ -652,15 +673,14 @@ Agent::doPreprocess()
     const int count_thr = ( wm.self().goalie()
                             ? 10
                             : 5 );
-    if ( wm.ball().posCount() > count_thr
+    if ( wm.ball().posCount() < count_thr
```

Comment:
This is a significant change the logic of the method. It should be submitted as a separate PR.